lib/buildkite/builder/extensions/sub_pipelines.rb

```diff
@@ -60,25 +67,14 @@ def prepare
         end
 
         dsl do
-          def pipeline(name, **options, &block)
+          def pipeline(name, &block)
             raise "Subpipeline must have a name" if name.empty?
             raise "Subpipeline does not allow nested in another Subpipeline" if context.is_a?(Buildkite::Builder::Extensions::SubPipelines::Pipeline)
 
             sub_pipeline = Buildkite::Builder::Extensions::SubPipelines::Pipeline.new(name, context, &block)
             context.data.pipelines.add(sub_pipeline)
 
-            options = options.slice(:key, :label, :async, :branches, :condition, :depends_on, :allow_dependency_failure, :skip, :emoji)
-            options[:key] ||= "subpipeline_#{name}_#{context.data.pipelines.count}"
-            options[:label] ||= name.capitalize
-
-            if options[:emoji]
-              emoji = Array(options.delete(:emoji)).map { |name| ":#{name}:" }.join
-              options[:label] = [emoji, options[:label]].compact.join(' ')
-            end
-
-            context.data.steps.add(Pipelines::Steps::Trigger, **options) do |context|
-              key context[:key]
-              label context[:label]
+            trigger_step = context.data.steps.add(Pipelines::Steps::Trigger) do
```

Comment:
Let’s remove the block form entirely otherwise it’s not clear why some are in the block and some are not.